src/mumble/ALSAAudio.cpp

```diff
@@ -519,60 +517,51 @@ void ALSAAudioOutput::run() {
 	qWarning("ALSAAudioOutput: Initializing %d channel, %d hz mixer", iChannels, iMixerFreq);
 	initializeMixer(chanmasks);
 
-	count = snd_pcm_poll_descriptors_count(pcm_handle);
-	snd_pcm_poll_descriptors(pcm_handle, fds, count);
-
 	qml.unlock();
 
 	while (bRunning && bOk) {
-		poll(fds, count, 20);
-		unsigned short revents;
-
-		snd_pcm_poll_descriptors_revents(pcm_handle, fds, count, &revents);
-		if (revents & POLLERR) {
-			snd_pcm_prepare(pcm_handle);
-		} else if (revents & POLLOUT) {
-			snd_pcm_sframes_t avail;
-			ALSA_ERRCHECK(avail = snd_pcm_avail_update(pcm_handle));
-			while (avail >= static_cast<int>(period_size)) {
-				stillRun = mix(outbuff, static_cast<int>(period_size));
-				if (stillRun) {
-					snd_pcm_sframes_t w = 0;
-					ALSA_RECOVER(pcm_handle, w=snd_pcm_writei(pcm_handle, outbuff, period_size));
-					if (w < 0) {
-						avail = w;
-						break;
-					}
-				} else
+		ALSA_RECOVER(pcm_handle, snd_pcm_wait(pcm_handle, 20));
```

Comment:
Should probably be followed by:

    if (err == 0) { // snd_pcm_wait timed out...
        continue;
    }